src/compiler/checker.ts

```diff
@@ -1412,6 +1411,14 @@ namespace ts {
                         return lastLocation.symbol;
                     }
                 }
+                if (originalLocation && isInJavaScriptFile(originalLocation) && originalLocation.parent) {

```

Comment:
There are 7 conditions here -- why put 3 in one `if` and 4 in another?

Reply:
Also, no need to test `&& originalLocation.parent` if we know it's an `Identifier`.

Reply:
Oops, that's a symptom of the development history. It looks a lot better with `isModuleExportsPropertyAccessExpression`.